os/board/common/partitions.c

```diff
@@ -317,9 +324,29 @@ void automount_fs_partition(partition_info_t *partinfo)
 		ret = mount(fs_devname, "/mnt", "smartfs", 0, NULL);
 		if (ret != OK) {
 			lldbg("ERROR: mounting '%s' failed, errno %d\n", fs_devname, get_errno());
+
+#ifdef CONFIG_BOARDCTL_RESET
+			/* Mount was failed, so we'll try low level format forcely */
+			lldbg("Low level format will be done, then board will reset\n");
+
+#ifdef CONFIG_SMARTFS_MULTI_ROOT_DIRS
+			ret = mksmartfs(fs_devname, 1, true);
+#else
+			ret = mksmartfs(fs_devname, true);
+#endif
+			if (ret != OK) {
+				lldbg("ERROR: mksmartfs on %s failed again\n", fs_devname);
+				return;
+			} 
+			/* After llformat, Reset to free all resources & initialize dev of smartfs */
```

Comment:
Is it alright to mention `llformat` here?
Since that is an internal function called at the lower level, maybe not known to everybody.
How about mentioning: -

> After mksmartfs, reset to free....

After reformatting, resetting the device should now be able to mount the smartfs partition successfully, right?